lib/Rx/Operator/MapOperator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+
+namespace Rx\Operator;
+
+
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+class MapOperator implements OperatorInterface
+{
+
+    /** @var callable  */
+    private $selector;
+
+    /**
+     * MapOperator constructor.
+     * @param $selector
+     */
+    public function __construct($selector)
+    {
+        if ( ! is_callable($selector)) {
```

Comment:
You can add a `callable` typehint to the constructor and remove this check as we're PHP 5.4+ now.